app/assets/javascripts/reducers/users.js

```diff
@@ -23,12 +25,16 @@ const SORT_DESCENDING = {
 
 export default function users(state = initialState, action) {
   switch (action.type) {
-    case RECEIVE_USERS: return {
-      ...state,
-      users: action.data.course.users,
-      isLoaded: true,
-      lastRequestTimestamp: Date.now()
-    };
+    case RECEIVE_USERS: {
+        const sortedUsers = sortByKey(action.data.course.users, state.sort.defaultKey, null, false, false, false);
```

Comment:
Using `defaultKey` here every time means it will still override the user-selected sort mode each time users are received. Instead, you could set `username` as the initial sortKey and then sort by the current sortKey when users are received.

Reply:
Yeah sure. Thank you. Let me update this.